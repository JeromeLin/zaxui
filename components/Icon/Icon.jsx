import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class Icon extends PureComponent {
  render() {
    const { prefixCls, type, theme, className } = this.props;
    const cls = classnames({
      [`${prefixCls}`]: true,
      [className]: !!className,
      [`${prefixCls}-${type}`]: !!type,
      [`theme-${theme}`]: !!theme,
    });

    return (
      <i className={cls} />
    );
  }
}

Icon.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.string,
  theme: PropTypes.oneOf(['default', 'primary', 'info', 'success', 'warning', 'error']),
};

Icon.defaultProps = {
  prefixCls: 'za-icon',
  className: null,
  type: '',
  theme: null,
};

export default Icon;
