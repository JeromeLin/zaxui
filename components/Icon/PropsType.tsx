export default interface PropsType {
  type?: string;
  theme?: 'default' | 'primary' | 'info' | 'success' | 'warning' | 'error';
  onClick?: () => void;
}
