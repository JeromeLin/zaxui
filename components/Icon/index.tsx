import React, { PureComponent } from 'react';
import { IconProps } from './PropsType';
import classnames from 'classnames';

export { IconProps };

export default class Icon extends PureComponent<IconProps, {}> {

  static defaultProps = {
    prefixCls: 'za-icon',
  }

  render() {
    const { prefixCls, type, theme, className, ...others } = this.props;
    const cls = classnames(prefixCls, className, {
      [`${prefixCls}-${type}`]: !!type,
      [`theme-${theme}`]: !!theme,
    });

    return (
      <i {...others} className={cls} />
    );
  }
}

