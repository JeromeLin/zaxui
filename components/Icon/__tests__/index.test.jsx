import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../index';

describe('Icon', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Icon type="right" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
