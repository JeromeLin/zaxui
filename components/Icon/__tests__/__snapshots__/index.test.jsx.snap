// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<i
  className="za-icon za-icon-right"
/>
`;
